javalib/src/main/scala/java/io/DataInputStream.scala

```diff
@@ -169,8 +185,15 @@ class DataInputStream(in: InputStream)
 
     def badFormat(msg: String) = throw new UTFDataFormatException(msg)
 
+    def getUnsignedByte(): Int =
+    try {
+    in.readUnsignedByte()
+    } catch {
+      case _: EOFException => -1
+    }
+
     while (i < length) {
-      val a = read()
+      val a = getUnsignedByte()
```

Comment:
What was wrong with `in.read()`?

Reply:
I changed the name of the method to `read` in the second commit because, after
tracing, I had more confidence that I would be using the intended local method.
To many `read()` methods floating around.

re: `What was wrong with in.read()`  

Let me describe my understanding, which may need correcting.

At that point in the code, `in` is given as a `DataInput`. As such,  it does not have a `read()` method
directly.  Some subclasses, which also follow `InputStream` might.  

The original code follows the "EOF indicated by returning -1" contract of `InputStream`.  The closest 
method in the `DataInput` interface is `readUnsignedByte()`. That is specified as throwing an 
exception on EOF.   The local `read()` method reduces required changes to the ported code by
translating the available `DataInput`  method to one which follows `InputStream`.

If there is ever a need in Scala.js for it to provide the static `readUTF()` method, the current 
approach, with suggested improvements, will make that easier.

Reply:
> At that point in the code, in is given as a DataInput. As such, it does not have a read() method
directly. Some subclasses, which also follow InputStream might.

Right. That is what I was missing.